client/my-sites/site-settings/form-general.jsx

```diff
@@ -127,35 +120,68 @@ const FormGeneral = React.createClass( {
 		return this.recordEventOnce.bind( this, key, eventAction );
 	},
 
+	setDirtyField( key ) {
+		const newState = {};
+		const dirtyFields = this.state.dirtyFields || [];
+		if ( dirtyFields.indexOf( key ) === -1 ) {
+			newState.dirtyFields = [ ...dirtyFields, key ];
+		}
+		this.setState( newState );
+	},
+
+	handleChange( event ) {
+		const currentTargetName = event.currentTarget.name,
+			currentTargetValue = event.currentTarget.value;
+
+		this.setDirtyField( currentTargetName );
+		this.setState( { [ currentTargetName ]: currentTargetValue } );
+	},
+
+	handleToggle( name ) {
+		return () => {
+			this.recordEvent.bind( this, `Toggled ${ name }` );
```

Comment:
Again, this has been borrowed from #10167, but it makes sense. I've removed it from this PR, but @ryelle might want to apply it to #10167 too.